### PR TITLE
[ci] Migrate all non-PQ CW340 tests to nightly CI

### PR DESCRIPTION
**Pull request by rtorok-zr**
CW340 runner wait time has been long due to limited hardware bandwidth, so this PR migrates the non-PQ CW340 tests to the nightly CI to speed up PR approvals.

### CODE REVIEW

File: .github/workflows/egret-cw340-pr.yml [hunk 1]
```diff
@@ -103,7 +23,6 @@ jobs:
       board: cw340
       interface: cw340
       tag_filters: cw340_pqc_test_rom
-      timeout_filters: 'short,moderate,long,eternal'
```
Comment (rtorok-zr) [inline .github/workflows/egret-cw340-pr.yml:106]: Question for @mkannwischer : Was it intentional the PQ tests were configured to run in the on-demand tests, even those with timeout > `moderate`?

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -26,6 +26,16 @@ jobs:
     uses: ./.github/workflows/ci.yml
     secrets: inherit
 
+  # Build non-PQC CW340 bitstream
+  chip_egret_cw340:
```
Comment (rtorok-zr) [inline .github/workflows/nightly.yml:30]: Aside: we should test if the existing CW340 tests work on the PQC-enabled bitstream. If they do, we could probably remove the non-PQ one from the CI entirely.

Comment (mkannwischer) [reply .github/workflows/nightly.yml:30]: They should work just fine, but I haven't tested it yet.
I just didn't want to touch the other tests and not duplicate any tests.

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -26,6 +26,16 @@ jobs:
...
+    name: Egret for CW340
+    needs: quick_lint
```
Comment (mkannwischer) [inline .github/workflows/nightly.yml]: I don't think this works because `quick_lint`  is defined in the other workflow.

File: .github/workflows/nightly.yml
Comment (mkannwischer) [inline .github/workflows/nightly.yml]: does this need a `needs: chip_egret_cw340` so it only starts after the bitstream is built?

File: .github/workflows/egret-cw340-pr.yml (hunk 1 above)
Comment (mkannwischer) [reply .github/workflows/egret-cw340-pr.yml:106]: The PQC tests have a `long` timeout, so I'm afraid that's needed. 
With this change this job will be a no-op.

File: .github/workflows/egret-cw340-pr.yml
Comment (mkannwischer) [inline .github/workflows/egret-cw340-pr.yml:32]: I'm surprised you didn't have to modify the verify-fpga job. How does this not complain about the missing CW340 runs?

File: .github/workflows/egret-cw340-nightly.yml [hunk 4]
```diff
@@ -102,7 +102,7 @@ jobs:
     secrets: inherit
     with:
       name: CW340 PQCEn=1 Long Tests
-      job_name: execute_long_pqc_fpga_tests_cw340
+      job_name: execute_pqc_fpga_tests_cw340
```
Comment (mkannwischer) [inline .github/workflows/egret-cw340-nightly.yml]: Looks like an accidental edit? I believe we need to keep that job_name unique.

File: .github/workflows/egret-cw340-nightly.yml [hunk 5]
```diff
@@ -12,88 +12,88 @@ on:
...
+      job_name: execute_manuf_fpga_tests_cw340
       bitstream: ${{ inputs.bitstream }}
       board: cw340
       interface: cw340
       tag_filters: "manuf,-cw310"
-      timeout_filters: 'long,eternal'
+      timeout_filters: 'short,moderate,long,eternal'
       timeout: 300
 
   execute_long_pqc_fpga_tests_cw340:
```
Comment (mkannwischer) [inline .github/workflows/egret-cw340-nightly.yml]: Thinking about it more, this tests just duplicates the per-PR one. 
Let's remove the job from the nightly and keep the `long` timeout filter in the PR?

**Review by mkannwischer — CHANGES_REQUESTED** [review]
Thanks @rtorok-zr for starting this! I support this change and think we should merge it as soon as possible. 
If I understand correctly, a couple of things still need to change before this will work as intended.

File: .github/workflows/egret-cw340-pr.yml
Comment (rtorok-zr) [reply .github/workflows/egret-cw340-pr.yml:32]: It will, I fixed this just now.

File: .github/workflows/egret-cw340-nightly.yml (hunk 4 above)
Comment (rtorok-zr) [reply .github/workflows/egret-cw340-nightly.yml]: I removed this job, per your other comment.

File: .github/workflows/nightly.yml [hunk 6]
```diff
@@ -39,7 +48,7 @@ jobs:
   # CW340 nightly FPGA jobs
   execute_nightly_tests_egret_cw340:
     name: Egret CW340 Nightly Tests
-    needs: on-demand
+    needs: chip_egret_cw340
```
Comment (mkannwischer) [inline .github/workflows/nightly.yml]: With this change I think it can happen that the nightly's `Verify` job starts before on-demand has finished and then it would fail. Easiest would be stick to on-demand here.

**Review by mkannwischer — APPROVED** [review]
Thanks @rtorok-zr. I think there is one race condition left, but the rest looks good to me.

File: .github/workflows/nightly.yml (hunk 6 above)
Comment (rtorok-zr) [reply .github/workflows/nightly.yml]: Ah, it should be 

```
needs:
- on-demand
- chip_egret_cw340
```
